src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -415,3 +387,42 @@ def _fetch_studio_specs(self, model_specs: JumpStartModelSpecs) -> Dict[str, Any
             Bucket=utils.get_jumpstart_content_bucket(self.region), Key=key
         )
         return json.loads(response["Body"].read().decode("utf-8"))
+
+
+    def scan_and_tag_models(self) -> None:
+        """Scans the Hub for JumpStart models and tags the HubContent.
+        
+        If the scan detects a model is deprecated or vulnerable, it will tag the HubContent.
+        The tags that will be added are based off the specifications in the JumpStart public hub:
+        1. "deprecated_versions" -> If the public hub model is deprecated
+        2. "inference_vulnerable_versions" -> If the public hub model has inference vulnerabilities
+        3. "training_vulnerable_versions" -> If the public hub model has training vulnerabilities
+
+        The tag value will be a list of versions in the Curated Hub that fall under those keys. 
+        For example, if model_a version_a is deprecated and inference is vulnerable, the
+        HubContent for `model_a` will have tags [{"deprecated_versions": [version_a]}, 
+        {"inference_vulnerable_versions": [version_a]}]
+        """
+        JUMPSTART_LOGGER.info(
+            "Tagging models in hub: %s", self.hub_name
+        )
+        js_models_in_hub = [model for model in self.list_models() if get_jumpstart_model_and_version(model) is not None]
+        tags_added: Dict[str, List[CuratedHubTag]] = {}
+        for model in js_models_in_hub:
+            tags_to_add: List[CuratedHubTag] = find_jumpstart_tags_for_hub_content(
+                hub_name=self.hub_name,
+                hub_content_name=model.hub_content_name,
+                region=self.region,
+                session=self._sagemaker_session
+            )
+            tag_hub_content(
+                hub_content_arn=model.hub_content_arn,
+                tags=tags_to_add,
+                session=self._sagemaker_session
+            )
+            tags_added.update({model.hub_content_arn: tags_to_add})
+
+        output_string = "No tags were added!"
+        if len(tags_added) > 0:
+          output_string = f"Added the following tags: {tags_added}"
```

Comment:
I still don't think this is the right call for logging tags. It would be better to contextualize the info and log when certain models in the Hub have been found to be unsupported. Can we please move this to inside `tag_hub_content`?